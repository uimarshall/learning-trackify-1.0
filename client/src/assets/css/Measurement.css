.measurement-circle {
  width: 60px;
  height: 60px;
}

.measurement-date {
  align-items: center !important;
  color: var(--clr-dark-grey) !important;
  font-weight: bold;
  font-size: 1.5rem !important;
}

.measurement-right-menu {
  display: flex;
  align-items: center !important;
  margin-left: auto;
  color: var(--clr-dark-grey) !important;
}

.measurements-title span {
  /* font-family: 'Helvetica Neue Bold' !important; */
  font-weight: bold;
  font-size: 0.8rem;
  color: var(--clr-grey-9) !important;
  margin-left: 10px;
}

.measurement-right-menu span:nth-child(1) {
  /* font-family: 'Helvetica Neue Bold' !important; */
  font-weight: bold;
  font-size: 1.3rem;
}

.measurement-right-menu span:nth-child(2) {
  margin-left: 5px;
  color: var(--clr-grey-9) !important;
  font-size: 0.7rem;
}

.measurement-right-menu a {
  margin-left: 5px;
  color: var(--clr-dark-grey);
  text-decoration: none;
  font-size: 1.5rem;
}

.measurements-title {
  margin: 8px;
}

.add-measure-button {
  position: absolute;
  right: 10px;
  bottom: 10px;
}

.add-measure-button a {
  color: white;
}

.last-div {
  padding-bottom: 11vh !important;
}

.new-measure {
  height: 80vh;
}

.new-measure select {
  color: #888;
  padding-left: 15px;
  height: 45px;
  width: 80%;
  max-width: 500px;
  border-radius: 6px;
  border: solid 1px #e8e8e8;
  background-color: #fff;
}

.measure-date {
  padding-top: 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.measure-date input {
  color: #888;
  padding-left: 15px;
  height: 45px;
  width: 80%;
  max-width: 500px;
  border-radius: 6px;
  border: solid 1px #e8e8e8;
  background-color: #fff;
}

.circle {
  padding: 10px;
  font-size: 5rem;
  display: flex;
  justify-content: center;
}

.buttons {
  display: flex;
  justify-content: space-around;
}
