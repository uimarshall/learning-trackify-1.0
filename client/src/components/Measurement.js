import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Circle from 'react-circle';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Time from '../helpers/Time';
import '../assets/css/Measurement.css';

export default function Measurement({
  measurement: {
    units, date_m: dateM, id, course_id: subjectId,
  },
}) {
  const style = {
    font: 'bold 7rem Helvetica, Arial, sans-serif',
  };
  let progress = Math.floor((units * 100) / 60);
  progress = progress < 100 ? progress : 100;
  return (
    <Link to={`/courses/${subjectId}/measurements/${id}`} className="subject-list-item bg-white">
      <div className="measurement-circle">
        <Circle
          responsive
          progress={progress} // here 60 is the goal
          progressColor="#8EE289"
          textStyle={style}
          textColor="#102a42"
        />
      </div>
      <div className="measurement-date">
        <span>{Time.format(dateM)}</span>
      </div>
      <div className="measurement-right-menu">
        <span>{units}</span>
        <span>Minutes</span>

        <FontAwesomeIcon icon={['fas', 'angle-right']} />
      </div>
    </Link>
  );
}

Measurement.propTypes = {
  measurement: PropTypes.shape({
    id: PropTypes.number,
    units: PropTypes.number,
    date_m: PropTypes.string,
    course_id: PropTypes.number,
  }).isRequired,
};
