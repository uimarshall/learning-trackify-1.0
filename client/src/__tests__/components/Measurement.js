import '@testing-library/jest-dom';

import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Measurement from '../../components/Measurement';

jest.mock('../../components/Measurement', () => () => <div id="Measurement">Measurement</div>);
describe('Measurement Component', () => {
  test('renders Measurement in the DOM', () => {
    const { container } = render(<Measurement />);
    const mockComponent = container.querySelector('div#Measurement');

    expect(mockComponent).toBeInTheDocument();
    expect(mockComponent).not.toHaveTextContent(/^dragon$/); // to match the whole content
    expect(mockComponent).not.toHaveTextContent(/fire$/i); // to use case-insensitive match
    expect(mockComponent).toHaveTextContent('Measurement');
  });
});

describe('Measurement Component', () => {
  afterEach(cleanup);

  it('should take a snapshot for Measurement Component', () => {
    const asFragment = renderer.create(<Measurement />).toJSON()
    expect(asFragment).toMatchSnapshot();
  });
});