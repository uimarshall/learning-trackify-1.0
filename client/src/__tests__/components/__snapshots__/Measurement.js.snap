// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Measurement Component should take a snapshot for Measurement Component 1`] = `
<div
  id="Measurement"
>
  Measurement
</div>
`;
